v-next/hardhat-ignition/src/internal/tasks/deploy.ts

```diff
@@ -14,16 +35,289 @@ interface TaskDeployArguments {
+        console.log("Deploy cancelled");
+        return;
+      }
+    }
+
+    if (reset && process.env.HARDHAT_IGNITION_CONFIRM_RESET === undefined) {
+      const resetPrompt = await Prompt({
+        type: "confirm",
+        name: "resetConfirmation",
+        message: `Confirm reset of deployment "${deploymentId}" on chain ${chainId}?`,
+        initial: false,
+      });
+
+      if (resetPrompt.resetConfirmation !== true) {
+        console.log("Deploy cancelled");
+        return;
+      }
+    }
+  } else if (deploymentDir !== undefined) {
+    // since we're on hardhat-network
+    // check for a previous run of this deploymentId and compare instanceIds
+    // if they're different, wipe deployment state
+    const instanceFilePath = path.join(
+      hre.config.paths.cache,
+      ".hardhat-network-instances.json",
+    );
+    const instanceFileExists = await exists(instanceFilePath);
+
+    const instanceFile: {
+      [deploymentId: string]: string;
+    } = instanceFileExists ? require(instanceFilePath) : {};
+
+    const metadata = (await connection.provider.request({
+      method: "hardhat_metadata",
+    })) as { instanceId: string };
+
+    if (instanceFile[deploymentId] !== metadata.instanceId) {
+      await remove(deploymentDir);
+    }
+
+    // save current instanceId to instanceFile for future runs
+    instanceFile[deploymentId] = metadata.instanceId;
+    await ensureDir(path.dirname(instanceFilePath));
+    await writeJsonFile(instanceFilePath, instanceFile);
+  }
+
+  if (reset) {
+    if (deploymentDir === undefined) {
+      throw new HardhatError(
+        HardhatError.ERRORS.IGNITION.CANNOT_RESET_EPHEMERAL_NETWORK,
+      );
+    } else {
+      await remove(deploymentDir);
+    }
+  }
+
+  if (strategyName !== "basic" && strategyName !== "create2") {
+    throw new HardhatError(HardhatError.ERRORS.IGNITION.UNKNOWN_STRATEGY);
+  }
+
+  await hre.tasks.getTask("compile").run({ quiet: true });
+
+  const userModule = await loadModule(hre.config.paths.ignition, modulePath);
+
+  if (userModule === undefined) {
+    throw new HardhatError(HardhatError.ERRORS.IGNITION.NO_MODULES_FOUND);
+  }
+
+  let parameters: DeploymentParameters | undefined;
+  if (parametersInput === undefined) {
+    parameters = await resolveParametersFromModuleName(
+      userModule.id,
+      hre.config.paths.ignition,
+    );
+  } else if (
+    parametersInput.endsWith(".json") ||
+    parametersInput.endsWith(".json5")
+  ) {
+    parameters = await resolveParametersFromFileName(parametersInput);
+  } else {
+    parameters = await resolveParametersString(parametersInput);
+  }
+
+  const accounts = (await connection.provider.request({
+    method: "eth_accounts",
+  })) as string[];
+
+  const artifactResolver = new HardhatArtifactResolver(hre);
+
+  const executionEventListener = new PrettyEventHandler();
+
+  const strategyConfig = hre.config.ignition.strategyConfig?.[strategyName];
+
+  try {
```

Comment:
all this ledger related things can be removed

Reply:
Now they'll be better served by the user interruptions system

Reply:
To confirm, we should remove these ledger related code blocks now, and add a task for after `hardhat-ledger` becoming available to re-enable Ignition support?

Reply:
I have removed the ledger related code in this commit: https://github.com/NomicFoundation/hardhat/pull/6227/commits/e5dd73f45d5154bd5db8ea7982b7755581ccc29b

This included removing some tests. We will revisit ledger+Ignition integration during the port of `hardhat-ignition`. The task has been added here: https://github.com/NomicFoundation/hardhat/issues/5646